CRM/Mailing/Event/BAO/MailingEventForward.php

```diff
@@ -38,28 +38,21 @@ public static function &forward($job_id, $queue_id, $hash, $forward_email, $from
     }
 
     // Find the email address/contact, if it exists.
-
-    $contact = CRM_Contact_BAO_Contact::getTableName();
-    $email = CRM_Core_BAO_Email::getTableName();
-    $queueTable = CRM_Mailing_Event_BAO_MailingEventQueue::getTableName();
-    $job = CRM_Mailing_BAO_MailingJob::getTableName();
-
-    $dao = new CRM_Core_DAO();
-    $dao->query("
-                SELECT      $contact.id as contact_id,
-                            $email.id as email_id,
-                            $contact.do_not_email as do_not_email,
-                            $queueTable.id as queue_id
-                FROM        ($email, $job as temp_job)
-                INNER JOIN  $contact
-                        ON  $email.contact_id = $contact.id
-                LEFT JOIN   $queueTable
-                        ON  $email.id = $queueTable.email_id
-                LEFT JOIN   $job
-                        ON  $queueTable.job_id = $job.id
-                        AND temp_job.mailing_id = $job.mailing_id
-                WHERE       $queueTable.job_id = $job_id
-                    AND     $email.email = '" .
+    $dao = CRM_Core_DAO::executeQuery("
+                SELECT      contact.id as contact_id,
+                            email.id as email_id,
+                            contact.do_not_email as do_not_email,
+                            queue.id as queue_id
+                FROM        (civicrm_email email, job as temp_job)
```

Comment:
This is wrong, should be
```suggestion
                FROM        (civicrm_email email, civicrm_job as temp_job)
```
But it's interesting that tests didn't pick that up.